spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkUtils.java

```diff
@@ -0,0 +1,210 @@
+   * featuresToInclude or featuresToInclude list.
+   *
+   * @param inputSchema       schema of the received record.
+   * @param featuresToInclude features to be used for training/prediction.
+   * @param featuresToExclude features to be excluded when training/predicting.
+   * @param labelField        field containing the prediction values.
+   * @return feature list to be used for training/prediction.
+   */
+  public static List<String> getFeatureList(Schema inputSchema, String featuresToInclude, String featuresToExclude,
+                                            String labelField) {
+    List<String> fields = new LinkedList<>();
+    List<Schema.Field> inputSchemaFields = inputSchema.getFields();
+    List<String> excludeFeatures = new ArrayList<>();
+
+    if (!Strings.isNullOrEmpty(featuresToInclude)) {
+      fields = Lists.newArrayList(Splitter.on(',').trimResults().split(featuresToInclude));
+      for (String field : fields) {
+        Schema.Field inputField = inputSchema.getField(field);
+        if (inputField == null) {
+          throw new IllegalArgumentException(String.format("Field %s does not exists in the input schema", field));
+        }
+      }
+    } else {
+      if (!Strings.isNullOrEmpty(featuresToExclude)) {
+        excludeFeatures.addAll(Lists.newArrayList(Splitter.on(',').trimResults().split(featuresToExclude)));
+      }
+      for (Schema.Field field : inputSchemaFields) {
+        String fieldName = field.getName();
+        if (!fieldName.equals(labelField) && (excludeFeatures.isEmpty() || !excludeFeatures.contains(fieldName))) {
+          fields.add(fieldName);
+        }
+      }
+    }
+    return fields;
+  }
+
+  /**
+   * Get the getCategoricalFeatureInfo using the feature to cardinality mapping provided by the user.
+   *
+   * @param inputSchema        schema of the received record.
+   * @param featuresToInclude  features to be used for training/prediction.
+   * @param featuresToExclude  features to be excluded when training/predicting.
+   * @param labelField         field containing the prediction values.
+   * @param cardinalityMapping feature to cardinality mapping specified for categorical features.
+   * @return categoricalFeatureInfo for categorical features.
+   */
+  public static Map<Integer, Integer> getCategoricalFeatureInfo(Schema inputSchema, String featuresToInclude,
+                                                                String featuresToExclude, String labelField,
+                                                                String cardinalityMapping) {
+    List<String> featureList = getFeatureList(inputSchema, featuresToInclude, featuresToExclude, labelField);
+    Map<Integer, Integer> outputFieldMappings = new HashMap<>();
+
+    if (!Strings.isNullOrEmpty(cardinalityMapping)) {
+      for (String field : Splitter.on(',').trimResults().split(cardinalityMapping)) {
+        String[] value = field.split(":");
```

Comment:
Is the format of the `cardinalityMapping` as `key1:value1,key2:value2,...` ? If that's the case, better use the `MapSplitter`.

```java
Map<String, String> map = Splitter.on(',').trimResults().withKeyValueSeparator(':').split(cardinalityMapping);
```